### PR TITLE
Rhys/typeahead update

### PR DESCRIPTION
**Pull request by wheresrhys**

### CODE REVIEW

File: header/js/suggestion-list.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const React = require('react');
+
+export class SuggestionList extends React.Component {
```
Comment (i-like-robots) [inline header/js/suggestion-list.js]: can be stateless

File: header/js/typeahead-new.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+			this.viewAllItemInnerHTML = '<button type="submit" data-trackable="view-all">View All Results</button>';
+		}
+
+		this.delegate = new Delegate(this.container);
+		this.bodyDelegate = new Delegate(document.body);
+		this.suggest = this.suggest.bind(this);
+		this.onType = debounce(this.onType, 150).bind(this);
+		this.onDownArrow = this.onDownArrow.bind(this);
+		this.onSuggestionKey = this.onSuggestionKey.bind(this);
+		this.onSuggestionClick = this.onSuggestionClick.bind(this);
```
Comment (i-like-robots) [inline header/js/typeahead-new.js]: why not make this into a react component as well?

Comment (wheresrhys) [inline header/js/typeahead-new.js]: I don't know react that well, and would refactoring bits of the component that don't actually need to repaint benefit the user anyway?

File: header/js/suggestion-list.js (hunk 1 above)
Comment (i-like-robots) [inline header/js/suggestion-list.js]:
```js
export (props) => {
}
```

File: header/js/suggestion-list.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	render () {
+		const suggestions = this.state.suggestions || [];
```
Comment (i-like-robots) [inline header/js/suggestion-list.js]: Oh wait, you have state - this should be declared in the constructor

Comment (i-like-robots) [inline header/js/suggestion-list.js]: But they'll be passed down right? So should be props?

**Pull request merged**